### ISSUE TITLE
nnz of the result is too large - spicy 1.6.0

### ISSUE THREAD

**Issue by hahmad2008** [scipy/scipy#14137]
I still encountred this issue when I execute the following: 
`cs = cosine_similarity(sparsed_data.transpose())`

RuntimeError:

```
__mul__
    return self._mul_sparse_matrix(other)
  File "/usr/local/lib/python3.6/dist-packages/scipy/sparse/compressed.py", line 509, in _mul_sparse_matrix
    np.asarray(other.indices, dtype=idx_dtype))
RuntimeError: nnz of the result is too large
```

 I use spicy 1.6.0 and still have the same issue, Is this problem due to using python 3.6? it seems it was solved in spicy 1.6.0 in [this issue](https://github.com/scipy/scipy/issues/12495). But I still have the same problem. Any help, please?

**Comment by tupui** [comment]
Hi, thanks for reporting. It's difficult to help without a working example. Also, can you post the result of `import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info)`?

Yes, SciPy 1.6 requires Python 3.7. Could you try maybe?

**Comment by hahmad2008** [comment]
Thank you @tupui 

With **Python 3.6.9**  seems that **spciy 1.6.0** doesn't exist.
 
```
python -c “import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info)”
1.5.4 1.16.0 sys.version_info(major=3, minor=6, micro=9, releaselevel=‘final’, serial=0)
```

* I tried to install it with another environment that has **Python 3.7.6**
* I got this output:

> ERROR: Could not find a version that satisfies the requirement spicy==1.6.0 (from versions: 0.16.0)
> ERROR: No matching distribution found for spicy==1.6.0

And this is the output for the command on env with python 3.7

```
python -c "import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info)"
1.4.1 1.18.1 sys.version_info(major=3, minor=7, micro=6, releaselevel='final', serial=0)
```

**Comment by tupui** [comment]
> Thank you @tupui
> 
> With **Python 3.6.9** seems that **spciy 1.6.0** doesn't exist.

Yes it's normal, we don't support Python 3.6 anymore. See [NEP29](https://numpy.org/neps/nep-0029-deprecation_policy.html) for more info.

> ```
> python -c "import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info)"
> 1.4.1 1.18.1 sys.version_info(major=3, minor=7, micro=6, releaselevel='final', serial=0)
> ```

This is weird, it means that you installed SciPy 1.4.1 in your Python 3.7 environment? Your environment does not seem to be clean. You may want to check your PYTHONPATH.

**Comment by andyfaff** [comment]
Have you checked that you try to install scipy, not spicy?

**Comment by tupui** [comment]
> Have you checked that you try to install scipy, not spicy?

I don't know how I missed the typo... Thanks for noticing 😅

**Comment by hahmad2008** [comment]
Thank you @andyfaff & @tupui  
my fault 😅 let me try it with scipy 1.6.0 and get back to you to check if I still encounter this runtime error (nnz of the result is too large).

**Comment by tupui** [comment]
Apparently there is a `spicy` package on PyPi and it imports `scipy`. IMO this is bad (there should at least be a warning). I understand where it comes from, but now PyPi has a way around this issue.

cc @Carreau

**Comment by tupui** [comment]
@hahmad2008 any update on this? Otherwise I will close it.

**Comment by hahmad2008** [comment]
It is solved even with ptyhon 3.9. Thank you guys. will close this issue.
